### DISCUSSION TITLE
RAG option

### DISCUSSION THREAD

**Discussion by theDaRkMaN1984** [xtekky/gpt4free#1769] [Ideas]
Hey there! I think it isn't currently possible to use RAG with G4F, but can you implement it please?
Or am i dumb and didn't see the tree in front of the forest? :D

Greetings
DaRkMaN1984

**Comment by eminsk** [comment]
Hello! Actually, you are right, it is currently not possible to use RAG with GPT-4 (G4F). RAG (Retrieval-Augmented Generation) and GPT-4 are different models and cannot simply be combined directly. You can use RAG to extract information from large data sets and pass that information to GPT-4 to generate text based on that information. You can create a pipeline that combines their functionality to solve specific problems.

**Reply by theDaRkMaN1984** [reply]
Can you lead me to an guide for that please? Can't find anything i understand without being in the materia

**Reply by TheMaskedAvengers** [reply]
I was actually wondering if it's possible to use G4F to provide the claude backend in this google collab worksheet on RAG?

https://colab.research.google.com/github/SamHollings/llm_tutorial/blob/main/llm_tutorial_rag.ipynb

Has the API been reverse engineered so it's just a matter of pointing the script at the G4F API address running locally? Can anyone guide me as to what changes need made if this is do-able? As would like to play with it.

Similarly very interested in getting GPT4 or the claude models working with the devika project. Again not sure if I just point devika at the API address of G4F and make up an API Key myself at random or something?

All guidance for this rookie much appreciated!
